apps/cmp.c

```diff
@@ -2295,6 +2351,35 @@ static int save_free_certs(STACK_OF(X509) *certs,
     return n;
 }
 
+static int save_crl(X509_CRL *crl,
+                    const char *file, const char *desc)
+{
+    BIO *bio = NULL;
+    int res = 0;
+
+    if (file == NULL)
+        return 1;
+    if (crl != NULL)
+        CMP_info2("received %s, saving to file '%s'", desc, file);
```

Comment:
And if `crl == NULL`?

Should this fail in this case? Since this is not blocked, `PEM_write_bio_X509_CRL()` will be called with NULL.

Reply:
OK, it appears that _either_ `delete_file()` or `save_crl()` will be called based on `crl`'s value, which means that `crl` should never be `NULL`, so this check is not necessary.

Reply:
Right - so this check could be removed as in the current context it is just confusing.
I'd say even better would be to adapt the code as follows, to be on the safe side for the future:
```suggestion
    if (crl == NULL)
        return 0;
    CMP_info2("received %s, saving to file '%s'", desc, file);
```